diff --git a/tensorflow/lite/micro/debug_log.cc b/tensorflow/lite/micro/debug_log.cc
index 98231968..b18ee19a 100644
--- a/tensorflow/lite/micro/debug_log.cc
+++ b/tensorflow/lite/micro/debug_log.cc
@@ -1,4 +1,5 @@
-/* Copyright 2023 The TensorFlow Authors. All Rights Reserved.
+/* Copyright 2020 The TensorFlow Authors. All Rights Reserved.
+   Copyright 2022-2025 NXP
 
 Licensed under the Apache License, Version 2.0 (the "License");
 you may not use this file except in compliance with the License.
@@ -13,36 +14,33 @@ See the License for the specific language governing permissions and
 limitations under the License.
 ==============================================================================*/
 
-// Reference implementation of the DebugLog() function that's required for a
-// platform to support the TensorFlow Lite for Microcontrollers library. This is
-// the only function that's absolutely required to be available on a target
-// device, since it's used for communicating test results back to the host so
-// that we can verify the implementation is working correctly.
-// This function should support standard C/C++ stdio style formatting
-// operations. It's designed to be as easy as possible to supply an
-// implementation though. On platforms that have a POSIX stack or C library, it
-// can be written as a single call to `vfprintf(stderr, format, args)` to output
-// a string to the error stream of the console, but if there's no OS or C
-// library available, there's almost always an equivalent way to write out a
-// string to some serial interface that can be used instead. To add an
-// equivalent function for your own platform, create your own implementation
-// file, and place it in a subfolder with named after the OS you're targeting.
-// For example, see the Cortex M bare metal version in the
-// tensorflow/lite/micro/bluepill/debug_log.cc file.
-
 #include "tensorflow/lite/micro/debug_log.h"
 
 #ifndef TF_LITE_STRIP_ERROR_STRINGS
 #include <cstdio>
 #endif
 
+#ifndef TF_LITE_STRIP_ERROR_STRINGS
+#ifdef SDK_DEBUGCONSOLE
+extern "C" {
+#include "fsl_debug_console.h"
+}
+#endif /* SDK_DEBUGCONSOLE */
+#endif
+
 extern "C" void DebugLog(const char* format, va_list args) {
 #ifndef TF_LITE_STRIP_ERROR_STRINGS
   // Reusing TF_LITE_STRIP_ERROR_STRINGS to disable DebugLog completely to get
   // maximum reduction in binary size. This is because we have DebugLog calls
   // via TF_LITE_CHECK that are not stubbed out by TF_LITE_REPORT_ERROR.
-  vfprintf(stderr, format, args);
-#endif
+#if SDK_DEBUGCONSOLE == DEBUGCONSOLE_DISABLE /* Disable debug console */
+#elif SDK_DEBUGCONSOLE == DEBUGCONSOLE_REDIRECT_TO_SDK /* Select printf, scanf, putchar, getchar of SDK version. */
+  DbgConsole_Vprintf(format, args);
+#elif !defined(SDK_DEBUGCONSOLE) || (SDK_DEBUGCONSOLE == \
+    DEBUGCONSOLE_REDIRECT_TO_TOOLCHAIN) /* Select printf, scanf, putchar, getchar of toolchain. \ */
+  vprintf(format, args);
+#endif /* SDK_DEBUGCONSOLE */
+#endif /* TF_LITE_STRIP_ERROR_STRINGS */
 }
 
 #ifndef TF_LITE_STRIP_ERROR_STRINGS
diff --git a/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc b/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc
index 8a3ee7b6..fcfe2f2d 100644
--- a/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc
+++ b/tensorflow/lite/micro/tools/make/targets/xtensa_makefile.inc
@@ -39,7 +39,6 @@ PLATFORM_FLAGS = \
   -DTF_LITE_MCU_DEBUG_LOG \
   -DTF_LITE_USE_CTIME \
   --xtensa-core=$(XTENSA_CORE) \
-  -mcoproc \
   $(TARGET_ARCH_DEFINES) \
   -mlongcalls

